### ISSUE TITLE
LiveServer not 

### ISSUE THREAD

**Issue by BolajiAyodeji** [ritwickdey/vscode-live-server#236]
## I'm submitting a...

```html
[x] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Feature request
[ ] Documentation issue or request
[ ] Other: <!-- Please describe: -->
```

## Current behavior

Live Server does not reload changes anymore, It only reloads and still show previous view, I would have to reload manually to get my new changes showing.

## Expected behavior

Once I return to the browser, server ought to reload and show new changes

## Environment

```html
Browser:

- [x] Chrome (desktop) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] IE version XX
- [ ] Edge version XX

For Tooling issues:

- Live Server: 5.2.0 <!-- Check which version is installed --> 
- Platform:  Windows <!-- Mac, Linux, Windows -->
- Visual Studio Code: 1.22.2 <!-- `code --version` -->
```

**Comment by shankarramr** [comment]
I am having the same issue.
Chrome Canary Version 72.0.3594.0 
VS Code Insider: 1.30.0
Live Server: 5.2.0

**Comment by ritwickdey** [comment]
Can anyone send a screenshot of Vscode console? (help -> Toggle developer tools)

**Comment by BolajiAyodeji** [comment]
@ritwickdey would do that now!

**Comment by BolajiAyodeji** [comment]
I just updated VsCode to the Latest version and it seems to be working fine.
Although it works for a while and stops, thrn I have to refresh again.
The auto reloading is not consistent.
I'm sending the  screenshot of VsCode console now.

**Comment by BolajiAyodeji** [comment]
![y](https://user-images.githubusercontent.com/30334776/48829266-31103300-ed72-11e8-916d-16b543dfba2f.PNG)

![yy](https://user-images.githubusercontent.com/30334776/48829276-379eaa80-ed72-11e8-9e83-e321f2afa140.PNG)

**Comment by BolajiAyodeji** [comment]
@ritwickdey

**Comment by ritwickdey** [comment]
On every change you should see, "Change detected" ? 

Let me know if you're getting `Change detected` but browser is not refreshing!! @BolajiAyodeji

**Comment by ritwickdey** [comment]
I just checked in my laptop. It's working fine.

**Comment by BolajiAyodeji** [comment]
`Change not detected` browser loading.
Its working fine here now on Latest VsCode update, but its not consistent, works fine for a while and stops reloading changes.
On older version it doesn't work at all
@ritwickdey

**Comment by BolajiAyodeji** [comment]
Alright!
Right now, Its very slow in loading.
Might take 3 seconds to load new changes.

Would take a shot once it stops working @ritwickdey

**Comment by BolajiAyodeji** [comment]
> > but its not consistent, works fine for a while and stops reloading changes.
> 
> When it stop to work, can take a screenshot of browser console and Vscode console?

Alright!
Right now, Its very slow in loading.
Might take 3 seconds to load new changes.

Would take a shot once it stops working @ritwickdey

**Comment by talinas** [comment]
Same here on latest public VScode and Chrome (actually tried FF and IE too). Both console tabs show nothing.

![image](https://user-images.githubusercontent.com/2170661/49301976-84028c80-f4ce-11e8-851e-e16d37eeb261.png)

**Comment by talinas** [comment]
Same here on latest public VScode and Chrome (actually tried FF and IE too). Both console tabs show nothing.

![image](https://user-images.githubusercontent.com/2170661/49302143-e65b8d00-f4ce-11e8-8796-5385f376d7a9.png)

**Comment by ritwickdey** [comment]
hi @talinas, You have not save the file. You've to save the file in order to live reload

**Comment by talinas** [comment]
Hi @ritwickdey,

Thanks for a quick reply! I did save it, but probably over clicked something additionally while doing screenshot, therefore vscode shows '1 unsaved', anyway sending one more screen with all saved now, but page not refreshed: 

![image](https://user-images.githubusercontent.com/2170661/49302846-d93f9d80-f4d0-11e8-8a9f-d748ffbb1483.png)

**Comment by ritwickdey** [comment]
Can you please open VsCode console (Help => Toggle Developer Tools)

On every saves you should see, "Change detected" ....

**Comment by ritwickdey** [comment]
And Can you share a screenshot of `View Source` Page?

**Comment by talinas** [comment]
I do see "change detected" events:
![image](https://user-images.githubusercontent.com/2170661/49303580-bb733800-f4d2-11e8-9054-239dd68e49cc.png)

![image](https://user-images.githubusercontent.com/2170661/49303662-f1182100-f4d2-11e8-8e76-66208f2f36f5.png)

**Comment by ritwickdey** [comment]
Oh. I got the issue. Valid JS tag is  `<script src="..."></script>` not `<script src="..." />`

https://www.w3schools.com/tags/att_script_src.asp

**Comment by talinas** [comment]
Whoops! That solved it, refresh now works well.

Thanks @ritwickdey !

**Comment by dalDevelo** [comment]
> hi @talinas, You have not save the file. You've to save the file in order to live reload

Indeed, it would be great if this could be reworked to refresh 'live', just as Brackets does it. #228 & #190. The extension is called 'live' server, after all 😁

**Comment by WadeShadow** [comment]
Sorry, guys, but it doesnt actually see my changes at all. Here is my vscode console:
![image](https://user-images.githubusercontent.com/44256364/54852958-0b327500-4cf7-11e9-9369-dcd290328a32.png)
No "changes detected" after any possible changes in css/html (saved file via different possible ways: interface, ctrl+S (if it has any meaning))

**Comment by Chandu4221** [comment]
Live Server Not Reloading the web page. 

My Config.
![ref1](https://user-images.githubusercontent.com/17278323/55053832-71b2ec80-5083-11e9-9a04-9525e9d47c89.PNG)
![ref2](https://user-images.githubusercontent.com/17278323/55053837-77a8cd80-5083-11e9-8b58-2bd36b0666ae.png)
![ref3](https://user-images.githubusercontent.com/17278323/55053908-c2c2e080-5083-11e9-8095-fddc01eef9ad.PNG)

"Changes Detected but no live reloading"
I am using the latest version of VSCode.

Please Fix this issue ASAP.

**Comment by Djokili** [comment]
Update visual studio code, bottom left click on settings and "check for updates".
Make sure you saved changes.
Make sure you have no other programs running on the same port as vsc.
Make sure you have body or head tag. 
Live Reload is not possible without body or head tag.

**Comment by A86S** [comment]
So i have disabled **AutoSave** option and then refresh the html page. After that its working fine.

**Comment by SphoorthyN** [comment]
It seems to do the actual refreshing on its own if the **Autosave** option is **enabled**, rather than me having to save manually each and every time I want my changes to appear in the browser.

**Comment by n1kalai** [comment]
> 
> 
> On every change you should see, "Change detected" ?
> 
> Let me know if you're getting `Change detected` but browser is not refreshing!! @BolajiAyodeji

i have same problem :/ can you help please?

**Comment by binuxo** [comment]
Please am having the same problem of browser not reloading or refreshing after save. Here is the vs console log I got.
![consol](https://user-images.githubusercontent.com/48191417/70242489-366bd000-1771-11ea-90d0-8d3a11ebc2be.png)

Please, I really need to resolve this.
#ritwickdey

**Comment by n1kalai** [comment]
try to delete node_modules and install then.. npm i or yarn install

чт, 5 дек. 2019 г. в 18:11, binuxo <notifications@github.com>:

**Comment by wilsonrg** [comment]
Live Server não atualiza após salvar arquivo html via vscode

Versões:
- Live Server v5.6.1
- VSCode v1.42.1

Tentativas de reparo:

- Desativado "onSave" para "off"
- Reinstalado Live Server
- Restart no VSCode

- salvamento automático: afterDelay
- AutoSaveDelay: 1
- LiveServer Wait: 1

- Removido a pasta: .vscode/ritwickdey.liveserver-5.6.1

Alguém tem uma solução por favor?

**Comment by wilsonrg** [comment]
Meu Live Server voltou a funcionar após instalar outro plugin chamado Live Server Preview
Instalei, reiniciei o VSCode e testei o Live Server que funcionou corretamente.
Espero que outras pessoas possam ser ajudadas com esta solução estranha assim como eu fui.

**Comment by Srijan-CdGd** [comment]
browser not reloading or web page not refreshing after the save of the file, the file is auto saved but i have to refresh web page manually to see the changes even when i am in a live sever

**Comment by halisogluiskender** [comment]
> browser not reloading or web page not refreshing after the save of the file, the file is auto saved but i have to refresh web page manually to see the changes even when i am in a live sever

I have that same problem.

**Comment by moof-moof** [comment]
Same problem here: Browser started by Live Server never updates without a manual a reload. (I am not using Autosave in VSC editor.)

**Comment by aakashmt97** [comment]
Hey @ritwickdey, I just installed Live Server in VSCODE, but the problem is there is NO "Reload" option showing so that I can activate. I'm not able to run my HTML code
![liveserver](https://user-images.githubusercontent.com/64053786/81523332-170c4b00-9380-11ea-9635-7a5cb16d17d5.PNG)

**Comment by Erina2016** [comment]
![Screenshot (98)](https://user-images.githubusercontent.com/51952228/84399668-efc8c800-ac22-11ea-817c-2f2edb6502ac.png)

**Comment by Erina2016** [comment]
Doesn't reload changes

**Comment by Pixelat0r** [comment]
Seems to not be working with the latest update (v1.46.0) on vscode. It detects changes but doesn't refresh the browser (at least for chrome and firefox).

**Comment by sekti92** [comment]
I just tried the same thing, but doesn't reload changes in my vscode

**Comment by patarok** [comment]
I had the same problem, I will see if this is still there tomorrow, but i know, that on an other install, it was just ENabling autosave with a delay of app. 1000ms ... this can be found in the VS Code settings ... and remember you will want to try :
'Auto Save'
>'afterDelay'
with an 
'Auto Save Delay' of
>'500'
thats milliseconds. So if you activate that every 1/2 second your VS code automatically saves the file, change will get detected and so the preview will refresh. MAYBE... today I also experienced some strange behavior on a very old devbox with ubuntu 18.04 installed where it just would not refresh "no matter what"...
I will report tomorrow. what could be the reason, if I have the time to find out.

**Comment by vinaynkashyap** [comment]
Auto Save is off
Change is detected by Live Sass
CSS outputs but without changes

#HACK 

Go to your main.scss file where you import all the other scss files and follow the below steps.

1. Comment // @import 'file.sass';
2. Save 
3 Remove the comment @import 'file.sass';
4. Save again IT WORKS!!!!

So to see the changes every time I have to do this.

Any solution?

**Comment by noobato1** [comment]
**SOLVED** (For me at least)

In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

**Comment by mikefox17** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

Yup, this worked for me as well. I just changed it to a false value.

**Comment by sfad159357** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now 
It's working for me!
thanks!

**Comment by Pouria-R** [comment]
Hi guys. it's simple. if you go to setting you will see auto save is off. choose "after delay" and done.

**Comment by ccbrandev** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

THANK YOU!! I spent LOTS of wasted time... SET THIS TO 'FALSE' in VS settings...
"liveServer.settings.useWebExt": false,

**Comment by theking2** [comment]
That did not solve the problem for me in the Dev console I see:
```
[Extension Host] Change detected d:\Cloud\SBW Haus des Lernens\M101 Webauftritt erstellen und veröffentlichen LG20 NM RH - General\input01\watteninseln.html
console.ts:137 [Extension Host] Looking for parseable documents...
console.ts:137 [Extension Host] Found all parseable documents.
console.ts:137 [Extension Host] Parsing documents and looking for CSS class definitions...
console.ts:137 [Extension Host] Summary:
console.ts:137 [Extension Host]% 1 parseable documents found
console.ts:137 [Extension Host]% 0 CSS class definitions found
console.ts:137 [Extension Host]% 0 unique CSS class definitions found
console.ts:137 [Extension Host]% 0 failed attempts to parse. List of the documents:
console.ts:137 [Extension Host] 
```

**Comment by steelvelveteen** [comment]
None of the solutions above works for me, problem still persists. Not only using vscode's Live Server extension but also npm's live-server as well. I have to refresh the browser myself. Using brave browser.

vscode 1.49.3 Live Server 5.6.1

**Comment by vinayaka16041999** [comment]
Problem still persists.It only works one time when i opened vs code for first time.After that it dosent change automatically.
Tried enabling auto save- toDelay
Tried changing browsers
In debug log of vscode it shows  change detected but in browser logs it dosent show change detected

**Comment by ehsanajmal** [comment]
can anyone help me please?
i tried all the solution above here but no one is working for me :( 
the problem is that the live server is not working on saving the file, it only works when i go and refresh the browser page
i tried all the solution but no one working 
can anyone help me with this please 
thanks in advance

**Comment by ivancardozo11** [comment]
If you are here couse live server is not reloading on Brave browser but it is on Chrome , try to disable Brave shields (the brave icon that disable scripts and that stuff...)

It work for me :p

**Comment by captainbrosset** [comment]
In many cases, the link between vscode and the browser, which the extension uses to reload the browser when the file is saved in vscode, is broken by invalid HTML.
The extension injects a `<script>` tag at the bottom of your HTML document to establish that link. If there's some invalid HTML before that tag, it may cause it to be ignored by the browser, and therefore the live reload won't work until you fix your HTML to be valid again.

One example of this is:
```html
<script src="my-script.js" />
```
This is invalid and causes the `<script>` tag that live-server injects to not be taken into account, it should be:
```html
<script src="my-script.js"></script>
```

Another example:
```html
<div style="
```
If you open an attribute but never close it, then it will cause the rest of the HTML to be considered as part of this attribute, and therefore will prevent live-server from injecting its script correctly.

It would be nice if the extension could inject its script in a safer way, but in the meantime, do make sure your HTML is correctly formed.

**Comment by kingma-sbw** [comment]
I've found out that adding a <head></head> to the page solved the problem

**Comment by theking2** [comment]
If the html file is missing a head the code cannot be injeckted. add an empty 
`<head></head>`
And reloads work

**Comment by NaolGB** [comment]
A day later I decided to used Edge and it worked (it didn't work on Chrome and Firefox). I also set auto save to after delay.

**Comment by NaolGB** [comment]
> A day later I decided to used Edge and it worked (it didn't work on Chrome and Firefox). I also set autosave to after delay.

I installed the Edge update on 05/24/21, not sure when it was downloaded

**Comment by bryanfks-dev** [comment]
I'm facing the same problem too, but I noticed something unusual, when I inspect element on my web, there's no `<!--Code injected by live-server-->`, I'm pretty sure this part should be inside the html. But, strangely it auto-reload on my phone. This issue happened after I tried live server extension on chrome to auto-reload my PHP server. After that, the auto-reload for html wasn't working, even tho I already uninstalled the chrome extension, reinstalled the live-server extension on vs code, and set `Live Server › Settings: Use Web Ext` to `False`. Anyone can help?
![Capture](https://user-images.githubusercontent.com/75253013/124551566-eaa96d00-de64-11eb-8cce-44ea12662f7d.PNG)

**Comment by Shantanu9799** [comment]
How to fix live server not auto reloading issue

**Comment by GyllenhaalSP** [comment]
> I'm facing the same problem too, but I noticed something unusual, when I inspect element on my web, there's no `<!--Code injected by live-server-->`, I'm pretty sure this part should be inside the html. But, strangely it auto-reload on my phone. This issue happened after I tried live server extension on chrome to auto-reload my PHP server. After that, the auto-reload for html wasn't working, even tho I already uninstalled the chrome extension, reinstalled the live-server extension on vs code, and set `Live Server › Settings: Use Web Ext` to `False`. Anyone can help?
> ![Capture](https://user-images.githubusercontent.com/75253013/124551566-eaa96d00-de64-11eb-8cce-44ea12662f7d.PNG)

Had the same problem. Installed PHP Debug and PHP Intelephense and HTML reloading would stop working. Uninstalled it and it seems to reload on save as usual.

**Comment by bryanfks-dev** [comment]
> > I'm facing the same problem too, but I noticed something unusual, when I inspect element on my web, there's no `<!--Code injected by live-server-->`, I'm pretty sure this part should be inside the html. But, strangely it auto-reload on my phone. This issue happened after I tried live server extension on chrome to auto-reload my PHP server. After that, the auto-reload for html wasn't working, even tho I already uninstalled the chrome extension, reinstalled the live-server extension on vs code, and set `Live Server › Settings: Use Web Ext` to `False`. Anyone can help?
> > ![Capture](https://user-images.githubusercontent.com/75253013/124551566-eaa96d00-de64-11eb-8cce-44ea12662f7d.PNG)
> 
> Had the same problem. Installed PHP Debug and PHP Intelephense and HTML reloading would stop working. Uninstalled it and it seems to reload on save as usual.

Yeah.. I already tried it and it worked. So, I recommend for php web development, it would be good if you use 3 packets in one application (Apache ,PHP, Mysql), for the example XAMPP and don't forget to turn `Live Server › Settings: Use Web Ext` to `true`. But, it couldn't auto-reload on phone. But, at least you have fully auto-reload for pc both for html+css and php (and auto-reload on phone only for html+css).

**Comment by bryanfks-dev** [comment]
Oh and, don't forget to download the live server web extension.

**Comment by Shantanu9799** [comment]
my problem was that unfortunately i was using another folder under my real
folder.. Now, it's ok......

**Comment by INGELM** [comment]
> 
> 
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

This working for me today! Thank you a lot!

**Comment by korayctn** [comment]
### SOLVED
Guys I figured out that if you have folder that is named with **uppercase letters**, live server doesn't refresh on the browser. You can solve this problem **by changing the folder name**.( at least it worked for me)

**Comment by aaronaan** [comment]
My Problem was the file was located in the ".vscode" folder
".../Downloads" works again

**Comment by pykeras** [comment]
solved by changing `fs.inotify.max_user_watches` from default value

more info at: [VSCODE docs](https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc)

**Comment by saikun0293** [comment]
> **SOLVED** (For me at least)
> 
> In settings search for "useWebExt" en check you DON'T have it enabled. I just unchecked this and now it's working!

This method is working for me. I guess it got enabled cause I was using live server extension for working on my PHP project earlier

**Comment by munaimpro** [comment]
Thanks, For me, its working after changing "Use Web Ext" to false! 
My VS Code version - 1.70.0, OS - Windows 10

**Comment by nacs1985** [comment]
> 

Work for me, 
the rest didn't work for me,
I changed the Uppercase to lowercase and it works right away
